src/coord/src/coord/dataflow_builder.rs

```diff
@@ -137,7 +141,11 @@ impl<'a> DataflowBuilder<'a> {
                             // If this source relies on any pre-existing indexes (i.e., indexes
                             // that we're not building as part of this `DataflowBuilder`), we're
                             // attempting to reinstantiate a single-use source.
-                            let intersection = self.indexes.intersection(dependent_indexes);
+                            let intersection = dependent_indexes
+                                .into_iter()
+                                .filter(|id| self.compute.collection(*id).is_ok())
+                                .collect::<Vec<_>>();
+
```

Comment:
As above, this behavior now certainly does not reflect any indexes that would have been inserted but delayed transmitting their creation commands to the controller. I .. can't speak to the reasoning about single-use sources though, and I don't know if this now produces false positives or negatives in those cases.